src/environmentd/src/http.rs

```diff
@@ -871,20 +871,30 @@ async fn auth(
                     return Err(AuthError::MissingHttpAuthentication)
                 }
             };
-            User {
-                external_metadata: Some(ExternalUserMetadata {
-                    user_id: claims.user_id,
-                    admin: claims.is_admin,
-                }),
-                name: claims.email,
-            }
+            // Set up an external metadata channel that never updates.
+            //
+            // NOTE(benesch): Arguably we should handle HTTP requests the same
+            // way we handle pgwire sessions: refresh for as long as we have
+            // valid credentials, updating external metadata along the way, and
+            // terminating the request if the refresh fails and the token
+            // expires. We don't do that presently, because most HTTP requests
+            // are very short lived, but they can in fact block for an
+            // arbitrarily long time.
```

Comment:
@mjibson I added a comment to the code based on your question [here](https://github.com/MaterializeInc/materialize/pull/25409/files#r1498425287).